go.mod

```diff
@@ -2,6 +2,8 @@ module sigs.k8s.io/cluster-api
 
 go 1.23.0
 
+toolchain go1.24.1
```

Comment:
/hold

Not going to merge toolchain directives

(cc @chrischdi @fabriziopandini the dependabot issue is back, it should not add these directives)